Chat/Infrastructure/Repositories/ChatRepository.cs

```diff
@@ -0,0 +1,73 @@
+﻿using Domain.Entities;
+using Domain.Interfaces;
+using Infrastructure.Data;
+using Microsoft.EntityFrameworkCore;
+
+
+namespace Infrastructure.Repositories
+{
+    public class ChatRepository : IChatRepository
+    {
+
+        private readonly DatabaseContext _db;
+
+        public ChatRepository(DatabaseContext db)
+        {
+            _db = db;
+        }
+
+        public async Task CreateAsync(Chat chat)
+        {
+           await _db.Chats.AddAsync(chat);
+        }
+
+        public  void Delete(int id)
+        {
+            var user = _db.Chats.Where(chat => chat.Id == id).Single();
+           _db.Chats.Remove(user);
+        }
```

Comment:
You can create an object with the passed id and delete it directly. There will be no need to retrieve the user object in this case